subprojects/core/src/main/java/org/gradle/process/internal/worker/request/ResponseProtocol.java

```diff
@@ -17,13 +17,27 @@
 package org.gradle.process.internal.worker.request;
 
 import org.gradle.process.internal.worker.child.WorkerLoggingProtocol;
+import org.gradle.process.internal.worker.problem.WorkerProblemProtocol;
 
-public interface ResponseProtocol extends WorkerLoggingProtocol {
+/**
+ * Protocol for sending information back from a worker process to the daemon.
+ * <p>
+ * This protocol is extended with the {@link WorkerLoggingProtocol} (for sending log messages)
+ * and {@link WorkerProblemProtocol} (for sending {@link org.gradle.api.problems.Problem}s through).
```

Comment:
💅 We might also add some rationale for why these things are a part of `ResponseProtocol`.  (i.e. because these protocols are tied to the build operation whose lifecycle is tightly coupled to the request/response interaction). We do have part of an explanation in `WorkerAction`, but it might make sense to include an explanation here as well.  Just so the next person who stumbles on this little bit of subtlety will have something to help understand it.

Reply:
Added, PTAL!